src/posts/posts.controller.ts

```diff
@@ -0,0 +1,185 @@
+import { CurrentUser } from '../../common/decorators/current-user';
+import { UserEntity } from '../users/user.entity';
+import { UserRoles } from '../../common/decorators/user.roles';
+import { Role } from '../../common/enums/users.role';
+import { Response } from 'express';
+import {
+    ApiBody,
+    ApiConsumes,
+    ApiCreatedResponse,
+    ApiOkResponse,
+    ApiOperation,
+    ApiParam,
+    ApiQuery,
+    ApiTags,
+} from '@nestjs/swagger';
+import { PostEntity } from './post.entity';
+import { ApiFileResponse } from '../../common/swagger/swagger.decorators';
+import { FilesService } from '../files/files.service';
+import { Pagination } from 'nestjs-typeorm-paginate';
+import { Order } from '../../common/enums/pagination.order';
+import { OrderCondition } from '../../common/enums/order.condition';
+
+@ApiTags('Posts')
+@UseGuards(JwtAuthGuard, UsersRoleGuard)
+@Controller('posts')
+export class PostsController {
+    constructor(private readonly postsService: PostsService, private filesService: FilesService) {}
+
+    @ApiOperation({ summary: 'Create Post' })
+    @ApiCreatedResponse({
+        type: PostEntity,
+        description: 'Created Post Response',
+    })
+    @ApiConsumes('multipart/form-data')
+    @FilesInterceptor('file')
+    @ApiBody({
+        type: CreatePostDto,
+        description: 'Create Post Fields',
+    })
+    @Post()
+    async createPost(
+        @Body() body: CreatePostDto,
+        @UploadedFile(fileValidationPipe('image/jpeg')) file: Express.Multer.File,
+        @CurrentUser() currentUser: UserEntity,
+    ): Promise<PostEntity> {
+        return this.postsService.createPost(body, currentUser, file);
+    }
+
+    @ApiOperation({ summary: 'Get All Posts' })
+    @ApiOkResponse({
+        type: PostEntity,
+        description: 'Posts Response',
+    })
+    @ApiQuery({ name: 'page', type: 'number', required: false })
+    @ApiQuery({ name: 'limit', type: 'number', required: false })
+    @ApiQuery({ name: 'order', enum: Order, required: false, example: 'ASC' })
+    @ApiQuery({ name: 'condition', enum: OrderCondition, required: false, example: 'ID' })
+    @Get()
+    async findAll(
+        @Query('page', new DefaultValuePipe(1), ParseIntPipe) page: number = 1,
+        @Query('limit', new DefaultValuePipe(10), ParseIntPipe) limit: number = 10,
+        @Query('order') order: Order = Order.ASC,
+        @Query('condition') condition: OrderCondition = OrderCondition.ID,
+    ): Promise<Pagination<PostEntity>> {
+        limit = limit > 100 ? 100 : limit;
+        return this.postsService.findAll({ page, limit }, order, condition);
+    }
+
+    @ApiOperation({ summary: 'Get Post By Id' })
+    @ApiOkResponse({
+        type: PostEntity,
+        description: 'Post Response',
+    })
+    @ApiParam({
+        name: 'id',
+        description: 'Post`s Id',
+    })
+    @Get(':id')
+    async findOne(@Param('id') id: string): Promise<PostEntity> {
+        return this.postsService.findOne(+id);
```

Comment:
This works, but please use `parseInt()` ot `Number()` functions instead of `+` to make the code readable and clear.